Devadv 813 xamarin forms upgrade

biozal opened the pull request:
this PR is an update to the latest version of Xamarin Forms and fixes for Xamarin Android:

-Updated Core, iOS, Android, and UWP to Xamarin Forms 5.0
-Fixed Android CSPROJ errors
-Upgraded Android to AndroidX
-Change versions of iOS, Android, and UWP to latest supported versions.
-Ran tests against each platform to make sure they still work.
-Removed open source dialog framework and coded to use built in Xamarin Forms Dialog framework.
-Fixes issues in iOS with Transport Security errors
-Fixes Android HTTP package errors with Dependabot.

biozal committed:
update to Xamarin Forms 5.0

this updates the apps to Xamarin Forms 5.0, removes ACR and adds in Xamarin Forms Display Dialog

biozal committed:
fixes to remove system.net.http

biozal committed:
UWP Update

fixes to get UWP working with Xamarin Forms 5.0

rajagp commented:
@biozal  Can you also update prerequisites specified in https://github.com/couchbaselabs/mobile-travel-sample/blob/master/content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc as appropriate

biozal committed:
updated images

updated images for documentation per PR request from Priya Rajagopal

biozal commented:
> @biozal Can you also update prerequisites specified in https://github.com/couchbaselabs/mobile-travel-sample/blob/master/content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc as appropriate

I updated the images.  I saw nothing in the document that stated what version of things you should use - only the screen shots that show the package versions - so I updated the two screen shots.

rajagp commented:
Thanks @biozal - the new screenshots should help but I was referring to the Prerequisites section which lists VS 15.6(?)- Is that even the right version of VS (community edition)? Also,  perhaps we can use the opportunity to list any other pre-requisites that you think would be helpful (for instance, version of Xcode to run a compatible version of iOS simulator)

biozal committed:
updated document for VS versions

updated the document for Visual Studio versions and Windows 10 version

biozal committed:
updated with notes on M1, Android, iOS, and Hot Reloaded

updated with information per PR request

biozal commented:
> Thanks @biozal - the new screenshots should help but I was referring to the Prerequisites section which lists VS 15.6(?)- Is that even the right version of VS (community edition)? Also, perhaps we can use the opportunity to list any other pre-requisites that you think would be helpful (for instance, version of Xcode to run a compatible version of iOS simulator)

Added version information and fully break down of Visual Studio, iOS, Android, Mac, Windows, and Hot Reloaded information.

rajagp commented on content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc:
```diff
@@ -3,10 +3,25 @@
+=== iOS 
+For iOS development, we recommend the latest version of XCode 11 or higher.  
+
+=== Android
```
Android Studio version? 4+?

biozal replied:
added documentation for that and changed wording to hopefully make it more clear with links to Google's documentation.

biozal committed:
added android requirements

added the android requirements and fixed some wording.

biozal committed:
modified wording

fixed working to make it more clear
